### PR TITLE
ar71xx: add support for TP-Link TL-WA850RE v2

### PR DESCRIPTION
**Pull request by pepe2k**
The following patch adds support for **TP-Link TL-WA850RE v2** (**QCA9533** based).

### CODE REVIEW

File: tools/firmware-utils/src/tplink-safeloader.c [hunk 1]
```diff
@@ -427,6 +427,42 @@ static struct device_info boards[] = {
...
+	/** Firmware layout for the TL-WA850RE v2 */
+	{
+		.id     = "TLWA850REV2",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:TL-WA850RE,product_ver:2.0.0,special_id:55530000}\n"
+			"{product_name:TL-WA850RE,product_ver:2.0.0,special_id:00000000}\n"
+			"{product_name:TL-WA850RE,product_ver:2.0.0,special_id:55534100}\n",
+		.support_trail = '\x00',
```
Comment (heinzek) [inline tools/firmware-utils/src/tplink-safeloader.c]: Hi. You can add here some eu special_ids:
based on latest official EU firmware:
```
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:45550000}
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:4B520000}
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:42520000}
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:4A500000}
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:43410000}
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:41550000}
{product_name:TL-WA850RE,product_ver:2.0.0,special_id:52550000}
```

Comment (pepe2k) [inline tools/firmware-utils/src/tplink-safeloader.c]: @heinzek, done, thanks! I have also fixed commit message as is turned out that there is no way to access the device over SSH in vendor firmware.

**Comment by blogic** [comment]
merged, Thanks !

**Comment by FreifunkUFO** [comment]
nice thank you! i tried it yesterday with a tplink device and its working now :-)
we also have WA854REv2 (same, but without ethernet). maybe, according to version 1 differences between 850 & 854 there is only a different TPLINK_HWID?

**Comment by pepe2k** [comment]
> nice thank you! i tried it yesterday with a tplink device and its working now :-)

Thanks for the feedback!

> we also have WA854REv2 (same, but without ethernet). maybe, according to version 1 differences between 850 & 854 there is only a different TPLINK_HWID?

Both v2 versions (and v3/v4 in case of TL-WA850RE) use different firmware upgrade process and image validation. Take a look at [tplink-safeloader](https://github.com/lede-project/source/blob/master/tools/firmware-utils/src/tplink-safeloader.c).

Cheers,
Piotr

**Comment by n3ph** [comment]
Hey pepe2k!

Thanks for your work on this!
I also try to get a valid factory image for the WA854REv2. How do I get these IDs?
https://github.com/lede-project/source/blob/master/tools/firmware-utils/src/tplink-safeloader.c#L436-L445

Best regrads,

n3ph

**Comment by heinzek** [comment]
> I also try to get a valid factory image for the WA854REv2. How do I get these IDs?
Check firmware file from official website. Firmware WA854REv2 EU have only one special id
from offset 0x3517ED
`SupportList:.{product_name:TL-WA854RE,product_ver:2.0.0,special_id:00000000}`

**Comment by FreifunkUFO** [comment]
so @n3ph and me were now able to test firmware on a WA854REv2, LEDE is working on a first device. :-)
my quick&dirty codechanges see https://github.com/FreifunkUFO/ledeprojectsource
of course, for testing wifi has to be turned on by default (f.i. via imagebuilder)

**Comment by ds2k5** [comment]
Hi,
 
Did i understand right...
 
download: git clone https://github.com/FreifunkUFO/ledeprojectsource

./scripts/feeds update -a

./scripts/feeds install -a

make menuconfig
 
select "AR9341"
 
and run "make" to build  ?

**Comment by mcoms** [comment]
Picked up one of these today, looking forward to `v17.01.2`. For now, I learn how to compile `master` 😄

**Pull request closed without merging**